From aea355abd6c87af54ad8c162731171e259e83f0d Mon Sep 17 00:00:00 2001
From: Michael Krummsdorf <michael.krummsdorf@tqs.de>
Date: Wed, 28 May 2014 14:41:58 +0200
Subject: [PATCH] ARM: TQMa28L: dt: Move e-MMC pinmux to module tree

Because this module has specific non-standard pinout
for the e-MMC on SSP1 controller, define these pins
in the module tree.

Signed-off-by: Michael Krummsdorf <michael.krummsdorf@tqs.de>
---
 arch/arm/boot/dts/imx28-mba28l.dts   |   18 ------------------
 arch/arm/boot/dts/imx28-tqma28l.dtsi |   20 ++++++++++++++++++++
 2 files changed, 20 insertions(+), 18 deletions(-)

diff --git a/arch/arm/boot/dts/imx28-mba28l.dts b/arch/arm/boot/dts/imx28-mba28l.dts
index 377a77d..df51b68 100644
--- a/arch/arm/boot/dts/imx28-mba28l.dts
+++ b/arch/arm/boot/dts/imx28-mba28l.dts
@@ -72,24 +72,6 @@
 					fsl,voltage = <1>;
 					fsl,pull-up = <0>;
 				};
-
-				mmc1_8bit_pins_a: mmc1-8bit@0 {
-					reg = <0>;
-					fsl,pinmux-ids = <
-						0x0001 /* MX28_PAD_GPMI_D00__SSP1_D0 */
-						0x0011 /* MX28_PAD_GPMI_D01__SSP1_D1 */
-						0x0021 /* MX28_PAD_GPMI_D02__SSP1_D2 */
-						0x0031 /* MX28_PAD_GPMI_D03__SSP1_D3 */
-						0x0041 /* MX28_PAD_GPMI_D04__SSP1_D4 */
-						0x0051 /* MX28_PAD_GPMI_D05__SSP1_D5 */
-						0x0061 /* MX28_PAD_GPMI_D06__SSP1_D6 */
-						0x0071 /* MX28_PAD_GPMI_D07__SSP1_D7 */
-						0x0151 /* MX28_PAD_GPMI_RDY1__SSP1_CMD */
-					>;
-					fsl,drive-strength = <1>;
-					fsl,voltage = <1>;
-					fsl,pull-up = <1>;
-				};
 			};
 
 			lcdif@80030000 {
diff --git a/arch/arm/boot/dts/imx28-tqma28l.dtsi b/arch/arm/boot/dts/imx28-tqma28l.dtsi
index 9255e62..4a180c3 100644
--- a/arch/arm/boot/dts/imx28-tqma28l.dtsi
+++ b/arch/arm/boot/dts/imx28-tqma28l.dtsi
@@ -29,6 +29,26 @@
 				vmmc-supply = <&reg_vddio_sd0>;
 				status = "okay";
 			};
+
+			pinctrl@80018000 {
+				mmc1_8bit_pins_a: mmc1-8bit@0 {
+					reg = <0>;
+					fsl,pinmux-ids = <
+						0x0001 /* MX28_PAD_GPMI_D00__SSP1_D0 */
+						0x0011 /* MX28_PAD_GPMI_D01__SSP1_D1 */
+						0x0021 /* MX28_PAD_GPMI_D02__SSP1_D2 */
+						0x0031 /* MX28_PAD_GPMI_D03__SSP1_D3 */
+						0x0041 /* MX28_PAD_GPMI_D04__SSP1_D4 */
+						0x0051 /* MX28_PAD_GPMI_D05__SSP1_D5 */
+						0x0061 /* MX28_PAD_GPMI_D06__SSP1_D6 */
+						0x0071 /* MX28_PAD_GPMI_D07__SSP1_D7 */
+						0x0151 /* MX28_PAD_GPMI_RDY1__SSP1_CMD */
+					>;
+					fsl,drive-strength = <1>;
+					fsl,voltage = <1>;
+					fsl,pull-up = <1>;
+				};
+			};
 		};
 	};
 
-- 
1.7.10.4

